app/actions/service_route_binding_delete.rb

```diff
@@ -4,9 +4,14 @@ class ServiceRouteBindingDelete
       class UnprocessableDelete < StandardError; end
 
       RequiresAsync = Class.new.freeze
-      DeleteComplete = Class.new.freeze
-      DeleteStarted = Struct.new(:operation).freeze
-      DeleteInProgress = Struct.new(:retry_after).freeze
+
+      DeleteStatus = Struct.new(:finished, :operation).freeze
+      DeleteStarted = ->(operation) { DeleteStatus.new(false, operation) }
+      DeleteComplete = DeleteStatus.new(true, nil).freeze
+
+      PollingStatus = Struct.new(:finished, :retry_after).freeze
+      PollingFinished = PollingStatus.new(true, nil).freeze
+      ContinuePolling = ->(retry_after) { PollingStatus.new(false, retry_after) }
```

Comment:
Starting to think we may want to extract this into a common place (create has similar structs)... But its fine as is for now